modules/http/src/main/java/org/mule/extension/http/api/listener/HttpBasicAuthenticationFilter.java

```diff
@@ -14,106 +14,60 @@
 import org.mule.runtime.module.http.internal.filter.BasicUnauthorisedException;
 import org.mule.service.http.api.domain.ParameterMap;
 
+import java.util.List;
+
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
 /**
- * Filter for basic authentication over an HTTP request
+ * Filter for basic authentication over an HTTP request.
+ *
+ * @since 4.0
  */
-public class HttpBasicAuthenticationFilter extends AbstractAuthenticationFilter {
+public class HttpBasicAuthenticationFilter {
 
   protected static final Log logger = LogFactory.getLog(HttpBasicAuthenticationFilter.class);
 
+  /**
+   * Authentication realm.
+   */
+  @Parameter
   private String realm;
 
-  private boolean realmRequired = true;
-  private HttpRequestAttributes attributes;
-
   /**
-   * Creates a filter based on the HTTP listener error response builder status code and headers configuration.
+   * The delegate-security-provider to use for authenticating. Use this in case you have multiple security managers defined in
+   * your configuration.
    */
-  public HttpBasicAuthenticationFilter() {
-    super();
-  }
-
-  @Override
-  protected void doInitialise() throws InitialisationException {
-    if (realm == null) {
-      if (isRealmRequired()) {
-        throw new InitialisationException(createStaticMessage("The realm must be set on this security filter"), this);
-      } else {
-        logger.warn("There is no security realm set, using default: null");
-      }
-    }
-  }
-
-  public String getRealm() {
-    return realm;
-  }
+  @Parameter
+  @Optional
+  @NullSafe
+  private List<String> securityProviders;
 
-  public void setRealm(String realm) {
-    this.realm = realm;
-  }
-
-  public boolean isRealmRequired() {
-    return realmRequired;
-  }
-
-  public void setRealmRequired(boolean realmRequired) {
-    this.realmRequired = realmRequired;
-  }
-
-  public void setAttributes(HttpRequestAttributes attributes) {
-    this.attributes = attributes;
-  }
-
-
-  protected Authentication createAuthentication(String username, String password) {
-    return new DefaultMuleAuthentication(new DefaultMuleCredentials(username, password.toCharArray()));
-  }
-
-  private Message createUnauthenticatedMessage() {
-    String realmHeader = "Basic realm=";
-    if (realm != null) {
-      realmHeader += "\"" + realm + "\"";
-    }
-    ParameterMap headers = new ParameterMap();
-    headers.put(WWW_AUTHENTICATE, realmHeader);
-    return Message.builder().nullPayload().attributes(new HttpListenerResponseAttributes(UNAUTHORIZED.getStatusCode(),
-                                                                                         UNAUTHORIZED.getReasonPhrase(),
-                                                                                         headers))
-        .build();
-  }
+  @Parameter
+  @Optional(defaultValue = "#[attributes]")
+  HttpRequestAttributes attributes;
```

Comment:
This is unfortunately required since it's been discussed and we can't change the basic auth listener implementation.